app/src/main/java/com/android/shelfLife/model/newhousehold/HouseHoldRepository.kt

```diff
@@ -6,6 +6,7 @@ interface HouseHoldRepository {
 
   val householdToEdit: StateFlow<HouseHold?>
   val households: StateFlow<List<HouseHold>>
+  val selectedHousehold: StateFlow<HouseHold?>
```

Comment:
You have moved the selectedHousehold variable from the userRepocitory to this one. You create the value here. The selected houshehold is null if no household is selected, that's why the type inside the stateFlow is HouseHold?